backend/src/install/cleanup.rs

```diff
@@ -233,12 +234,16 @@ pub async fn remove_from_current_dependents_lists<
     Ok(())
 }
 
-#[instrument(skip(ctx, db))]
-pub async fn uninstall(
+#[instrument(skip(ctx, secrets, db))]
+pub async fn uninstall<Ex>(
     ctx: &RpcContext,
     db: &mut PatchDbHandle,
+    secrets: &mut Ex,
     id: &PackageId,
-) -> Result<(), Error> {
+) -> Result<(), Error>
+where
+    for<'a> &'a mut Ex: Executor<'a, Database = Sqlite>,
```

Comment:
Comment: First time seeing a for<'a>
Most of the time I would have had in line 241 something like `secrets: & mut impl Executor<'_, Database = Sqlite>`
Looks like the for is a way of tying the lifetimes together.